unison-cli/src/Compat.hs

```diff
@@ -0,0 +1,45 @@
+{-# LANGUAGE CPP #-}
+
+module Compat where
+
+import Control.Concurrent (mkWeakThreadId, myThreadId)
+import Control.Exception (AsyncException (UserInterrupt), throwTo)
+import System.Mem.Weak (deRefWeak)
+import Unison.Prelude
+import qualified UnliftIO
+
+#if defined(mingw32_HOST_OS)
+import qualified GHC.ConsoleHandler as WinSig
+#else
+import qualified System.Posix.Signals as Sig
+#endif
+
+defaultInterruptHandler :: IO (IO ())
```

Comment:
We can likely come up with a more helpful top-level handler at some point, but this is what we had before 🤷🏼‍♂️